onnx/backend/test/case/node/optionalhaselement.py

```diff
@@ -21,43 +21,45 @@ class OptionalHasElement(Base):
-        input_type_proto = onnx.helper.make_optional_type_proto(tensor_type_proto)
-        node = onnx.helper.make_node(
-            'OptionalHasElement',
-            inputs=['optional_input'],
-            outputs=['output']
-        )
-        output = optional_has_element_reference_implementation(optional)
-        expect(node, inputs=[optional], outputs=[output],
-               input_type_protos=[input_type_proto],
-               name='test_optional_has_element')
+        optional_type_proto = onnx.helper.make_optional_type_proto(tensor_type_proto)
+
+        # OptionalHasElement takes a tensor or optional as input
+        for input_type_protos in [tensor_type_proto, optional_type_proto]:
+            node = onnx.helper.make_node(
+                'OptionalHasElement',
+                inputs=['optional_input'],
+                outputs=['output']
+            )
+            output = optional_has_element_reference_implementation(optional)
+            test_name = 'test_optional_has_element_' +\
+                ('optional_input' if input_type_protos == optional_type_proto else 'tensor_input')
+            expect(node, inputs=[optional], outputs=[output],
+                input_type_protos=[optional_type_proto],
+                name=test_name)
 
     @staticmethod
     def export_empty() -> None:
         optional = None
+
         tensor_type_proto = onnx.helper.make_tensor_type_proto(elem_type=onnx.TensorProto.INT32, shape=[])
-        input_type_proto = onnx.helper.make_optional_type_proto(tensor_type_proto)
-        node = onnx.helper.make_node(
-            'OptionalHasElement',
-            inputs=['optional_input'],
-            outputs=['output']
-        )
-        output = optional_has_element_reference_implementation(optional)
-        expect(node, inputs=[optional], outputs=[output],
-               input_type_protos=[input_type_proto],
-               name='test_optional_has_element_empty')
+        optional_type_proto = onnx.helper.make_optional_type_proto(tensor_type_proto)
 
-    @staticmethod
-    def export_tensor() -> None:
-        tensor = np.array([1, 2, 3, 4]).astype(np.float32)
-        tensor_type_proto = onnx.helper.make_tensor_type_proto(elem_type=onnx.TensorProto.FLOAT, shape=[4, ])
-        input_type_proto = onnx.helper.make_optional_type_proto(tensor_type_proto)
-        node = onnx.helper.make_node(
-            'OptionalHasElement',
-            inputs=['optional_input'],
-            outputs=['output']
-        )
-        output = optional_has_element_reference_implementation(tensor)
-        expect(node, inputs=[tensor], outputs=[output],
-               input_type_protos=[input_type_proto],
-               name='test_optional_has_element_tensor')
+        # OptionalHasElement takes a tensor or optional as input
+        for input_type_proto in [tensor_type_proto, optional_type_proto]:
+            input_name_options = {
+                'empty': 'optional_input',
```

Comment:
I think the combination of `input_type_proto == tensor_type_proto` and above (`'empty': 'optional_input'`) is not a valid combination to test? That might also be triggering the change to cmd_tools.py below?